File: examples/03_module/main.js
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Entry point. Orchestrate order to invoice.

import { readOrderFromFile } from "./order/order.js";
import { checkStockAvailability } from "./stock/stock.js";
import { createInvoiceFromOrder, writeInvoiceToFile } from "./billing/billing.js";

const order = readOrderFromFile("order.txt");

if (!checkStockAvailability(order.product, order.quantity)) {
  process.stderr.write(`out of stock: ${order.product}\n`);
  process.exit(1);
}

const invoice = createInvoiceFromOrder(order, 29.99);
writeInvoiceToFile(invoice, "invoice.txt");

File: examples/03_module/main.py
# Lean Code — BSD 3-Clause License — Vivian Voss, 2026
# Scope: Entry point. Orchestrate order to invoice.

import sys
from order.order import read_order_from_file
from stock.stock import check_stock_availability
from billing.billing import create_invoice_from_order, write_invoice_to_file

order = read_order_from_file("order.txt")

if not check_stock_availability(order["product"], order["quantity"]):
    sys.exit(f"out of stock: {order['product']}")

invoice = create_invoice_from_order(order, 29.99)
write_invoice_to_file(invoice, "invoice.txt")
